### PR TITLE
shuffles docs and add details about man-roxygen

### PR DESCRIPTION
**Pull request by maelle**
This is mainly to implement your notes from #72 but also a bit of reshuffling (putting that notes in the documentation subsection, and starting that subsection with roxygen2, since before it started with remarks about roxygen tags before introducing roxygen2)

### CODE REVIEW

File: pkg_building.Rmd [hunk 1]
```diff
@@ -197,8 +204,9 @@ Please do not list editors as contributors. Your participation in and contributi
 
 ## Examples
 
-* Include extensive examples in the documentation. In addition to demonstrating how to use the package, these can act as an easy way to test package functionality before there are proper tests. However, keep in mind we require tests in contributed packages. If you prefer not to clutter up code with extensive documentation, place further documentation/examples in files in a `man-roxygen` folder in the root of your package, and those will be combined into the manual file by the use of `@template <file name>`, for example. You can run examples with `devtools::run_examples()`.
+* Include extensive examples in the documentation. In addition to demonstrating how to use the package, these can act as an easy way to test package functionality before there are proper tests. However, keep in mind we require tests in contributed packages. 
 
+*  You can run examples with `devtools::run_examples()`.
```
Comment (sckott) [inline pkg_building.Rmd]: @maelle should we add that we want pkg maintainers to make sure they run examples wrapped in `\dontrun{}` or `\donttest{}` somewhere?  most likely on CI, either travis or appveyor or both. `r_check_args: "--as-cran --run-dontrun"`

**Comment by sckott** [comment]
good point, we should mention `@inherits` tags.

File: pkg_building.Rmd (hunk 1 above)
Comment (sckott) [reply pkg_building.Rmd]: @maelle what did you think about this comment?

**Comment by sckott** [comment]
either way, happy to do it, just let me know

**Comment by sckott** [comment]
other edits are 👍

**Comment by maelle** [comment]
Perfect if you do the other edits 😇

**Comment by maelle** [comment]
Thanks

File: pkg_building.Rmd (hunk 1 above)
Comment (maelle) [reply pkg_building.Rmd]: good point!

Comment (maelle) [reply pkg_building.Rmd]: can you add this point? Thanks!

Comment (sckott) [reply pkg_building.Rmd]: sure

**Comment by maelle** [comment]
Cool, thanks, I think we can merge now?

**Comment by sckott** [comment]
yeah

**Pull request merged**